drivers/ethernet/eth_stm32_hal.c

```diff
@@ -28,22 +28,137 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #include "eth_stm32_hal_priv.h"
 
 #if defined(CONFIG_ETH_STM32_HAL_USE_DTCM_FOR_DMA_BUFFER) && \
-	    !DT_NODE_HAS_STATUS(DT_CHOSEN(zephyr_dtcm), okay)
+			!DT_NODE_HAS_STATUS(DT_CHOSEN(zephyr_dtcm), okay)
 #error DTCM for DMA buffer is activated but zephyr,dtcm is not present in dts
 #endif
 
+static HAL_StatusTypeDef read_eth_phy_register(ETH_HandleTypeDef * heth,
+											   uint32_t PHYAddr,
+											   uint32_t PHYReg,
+											   uint32_t *RegVal);
+
+#ifndef PHY_ADDR
+#define PHY_ADDR        ((uint32_t)CONFIG_ETH_STM32_HAL_PHY_ADDRESS)
+#endif
+
+#ifndef PHY_BSR
+#define PHY_BSR     ((uint16_t)0x0001U)
+#endif
+
+#ifndef PHY_LINKED_STATUS
+#define PHY_LINKED_STATUS       ((uint16_t)0x0004U)
+#endif
+
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+/**
+ * Definitions for ethernet driver of all others STM32 except STM32H7
+ */
+
+#define GET_TX_DESC_LIST_PTR(heth)   (&(((ETH_HandleTypeDef *)heth)-> \
+																TxDescList))
+
+#define GET_FIRST_DMA_TX_DESC(heth)  ((ETH_DMADescTypeDef *) \
+									  (((ETH_HandleTypeDef *)heth)-> \
+																Init.TxDesc))
+#define GET_FIRST_DMA_RX_DESC(heth)  ((ETH_DMADescTypeDef *) \
+									  (((ETH_HandleTypeDef *)heth)-> \
+																Init.RxDesc))
+
+#define IS_ETH_DMATXDESC_OWN(dma_tx_desc)   (((ETH_DMADescTypeDef *) \
+												dma_tx_desc)->DESC3 & \
+												ETH_DMATXNDESCRF_OWN)
+#define GET_DMA_TX_BUFFER1_ADDR(dma_tx_desc) (((ETH_DMADescTypeDef *) \
+												dma_tx_desc)->DESC0)
+
+#define ETH_RXBUFNB                     ETH_RX_DESC_CNT
+#define ETH_TXBUFNB                     ETH_TX_DESC_CNT
+
+#define ETH_MEDIA_INTERFACE_MII         HAL_ETH_MII_MODE
+#define ETH_MEDIA_INTERFACE_RMII        HAL_ETH_RMII_MODE
+
+#define LL_AHB1_GRP1_PERIPH_ETHMAC      LL_AHB1_GRP1_PERIPH_ETH1MAC
+#define LL_AHB1_GRP1_PERIPH_ETHMACTX    LL_AHB1_GRP1_PERIPH_ETH1TX
+#define LL_AHB1_GRP1_PERIPH_ETHMACRX    LL_AHB1_GRP1_PERIPH_ETH1RX
+#define LL_AHB1_GRP1_PERIPH_ETHMACPTP   LL_AHB1_GRP1_PERIPH_ETH1MAC
+
+#define ETH_DMA_TRANSMIT_TIMEOUT        20U
+
+static HAL_StatusTypeDef read_eth_phy_register(ETH_HandleTypeDef *heth,
+											   uint32_t PHYAddr,
```

Comment:
can you check the indentation here and below, there are indent issues in so many places that I do not bother to comment them all

Reply:
done